src/autocomplete/index.js

```diff
@@ -0,0 +1,200 @@
+      }
+      if (!list.includes(selectedVal)) setList(newList);
+      changeFormValue(newList);
+      setFilteredSuggestions([
+        ...suggestions.filter(suggestion => !newList.includes(suggestion))
+      ]);
+    }
+    setQuery('');
+  };
+
+  const onKeyDown = e => {
+    const { onChange: changeFormValue } = input;
+    const { keyCode } = e;
+    if (keyCode) {
+      if (keyCode === 13 || keyCode === 9) addToList(keyCode);
+      if (keyCode === 38 || keyCode === 40) changeSelectedIndex(keyCode);
+      if (keyCode === 8 && list.length > 0 && query === '') {
+        let tmpList = [...list];
+        tmpList.splice(-1, 1);
+        setList(tmpList);
+        changeFormValue(tmpList);
+      }
+    }
+  };
+
+  const removeItem = position => () => {
+    const { onChange: changeFormValue } = input;
+    let newList = [...list].filter((item, index) => index !== position);
+    setList(newList);
+    changeFormValue(newList);
+  };
+
+  const showError = meta => meta.error && meta.touched;
+  const showWarning = meta => meta.error && meta.touched;
+
+  useEffect(() => {
+    if (inputEl.current) {
+      const pos = inputEl.current.getBoundingClientRect();
+      setTopOffset(pos.top + 5);
+      setLeftOffset(pos.left + 5);
+    }
+  }, [inputEl, list]);
+
+  return (
+    <Container>
+      <InputContainer
+        warning={showWarning(meta)}
+        active={meta.active}
+        disabled={disabled}
+        flexGrow={flexGrow}
+      >
+        {AddOn && (
+          <AddOnContainer sm={addOnSm}>
+            <AddOn />
+          </AddOnContainer>
+        )}
+
+        {list.map((item, index) => (
+          <Tag slim onRemove={removeItem(index)} key={`item-${index}`}>
+            {item}
+          </Tag>
+        ))}
+
+        <TextInput
+          disabled={disabled}
+          readOnly={readOnly}
+          {...input}
+          {...rest}
+          onChange={onChange}
+          value={query}
+          onKeyDown={onKeyDown}
+          ref={inputEl}
+        />
+
+        {Affix && (
+          <AffixContainer sm={addOnSm}>
+            <Affix />
+          </AffixContainer>
+        )}
+
+        {Icon && (
+          <IconRegion>
+            <Icon />
+          </IconRegion>
+        )}
+      </InputContainer>
+      {showError(meta) && <ErrorMessage>{meta.error}</ErrorMessage>}
+      {filteredSuggestions.length > 0 && query !== '' && (
+        <SuggestionList top={topOffset} left={leftOffset}>
+          {filteredSuggestions.map((suggestion, index) => (
+            <SuggestionItem
+              selected={index === selectedIndex}
+              key={`suggestion${index}`}
+            >
+              {suggestion}
+            </SuggestionItem>
+          ))}
+        </SuggestionList>
+      )}
+    </Container>
+  );
+};
+
+export default Autocomplete;
+
+Autocomplete.propTypes = {
+  input: PropTypes.object,
+  addOn: PropTypes.func,
+  affix: PropTypes.object,
+  icon: PropTypes.object,
```

Comment:
Generally `PropTypes.object` is frowned upon as it's a bit too vague.
Here I'd suggest using `PropTypes.node` for the icon which implies it needs a node to function correctly.
As supplying something like ```{ x: 123 }``` would break the component.

Reply:
will do. I copied the `input` component that is already in oxygen and just added on to it

Reply:
No problem, there's some tech debt that needs addressin in the project unfortunately.

I think you can safely ignore `input` here as it has a massive amount of properties to define and wouldn't make sense.

I'd just worry about `affix` and `icon`